ckanext/switzerland/dcat/harvesters.py

```diff
@@ -158,3 +188,146 @@ def before_update(self, harvest_object, dataset_dict, temp_dict):
                 resource['id'] = resource_mapping[identifier]
+
+    def gather_stage(self, harvest_job): # noqa
+
+        log.debug('In swiss dcat harvester gather_stage')
+
+        gather_config = self._set_gather_config(harvest_job.source.config)
+
+        # Get file contents of first page
+        next_page_url = harvest_job.source.url
+
+        guids_in_source = []
+        object_ids = []
+        last_content_hash = None
+        page_count = 0
+
+        while next_page_url:
+            next_page_url, before_download_errors = \
+                self.before_download(next_page_url, harvest_job)
+            page_count += 1
+
+            for error_msg in before_download_errors:
+                self._save_gather_error(error_msg, harvest_job)
+
+            if not next_page_url:
+                return []
+
+            content, rdf_format = self._get_content_and_type(
+                next_page_url, harvest_job, 1,
+                content_type=gather_config['rdf_format'])
+
+            content_hash = hashlib.md5()
+            if content:
+                content_hash.update(content)
+
+            if last_content_hash:
+                if content_hash.digest() == last_content_hash.digest():
+                    log.warning("Remote content was the same even when using a paginated URL, skipping") # noqa
+                    break
+            else:
+                last_content_hash = content_hash
+
+            content, after_download_errors = \
+                self.after_download(content, harvest_job)
+
+            for error_msg in after_download_errors:
+                self._save_gather_error(error_msg, harvest_job)
+
+            if not content:
+                return []
+
+            parser = SwissDCATRDFParser()
+
+            try:
+                # the harvest source is parsed
+                # when the shacl validation failed the result
+                # will be False
+                log.debug("SHACL parsing page {} shacl validation: {}"
+                          .format(page_count,
+                                  gather_config['shacl_validation']))
+                parser.parse(
+                    content,
+                    _format=rdf_format,
+                    harvest_source_id=harvest_job.source_id,
+                    harvest_job_id=harvest_job.id,
+                    page_count=page_count,
+                    shacl_validation=gather_config['shacl_validation'],
+                    shacl_file=gather_config['shacl_validation_file'],
+                )
+                if gather_config['shacl_validation']:
+                    # in case of shacl validation errors they are
+                    # reported as gather errors per page that was parsed
+                    msg_count = 0
+                    shacl_error_dict = \
+                        parser.shaclresults.errors_grouped_by_node()
+                    if shacl_error_dict:
+                        log.debug('SHACL error node count: {}'
```

Comment:
have you considered using this syntax? `{0}` It is used everywhere else and I don't see a clear reason to do it differently here